CONTRIBUTING.md

```diff
@@ -1,6 +1,143 @@
-# Contributing guidelines
+# Contributing to GLASS
 
-## Workflow
+If you are planning to develop _GLASS_, or want to use the latest commit of
+_GLASS_ on your local machine, you might want to install it from the source.
+This installation is not recommended for users who want to use the stable
+version of _GLASS_. The page below describes how to build, test, and develop
+_GLASS_.
+
+## Installation
+
+The developer installation of _GLASS_ comes with several optional dependencies -
+
+- `test`: installs `scipy`, `fitsio`, and the relevant `pytest` dependencies
+- `docs`: installs `sphinx`, `nbsphinx`, and a few other docs related
+  dependencies
+- `examples`: installs `camb`, `glass.ext.camb`, and a few notebook related
+  dependencies
```

Comment:
Do we want to hardcode this here? If we change dependencies, it is another thing to change. Could just say `test` is related to testing dependencies